dartagnan/src/main/java/com/dat3m/dartagnan/solver/caat4wmm/WMMSolver.java

```diff
@@ -0,0 +1,147 @@
+package com.dat3m.dartagnan.solver.caat4wmm;
+
+
+import com.dat3m.dartagnan.solver.caat.CAATSolver;
+import com.dat3m.dartagnan.solver.caat.reasoning.CAATLiteral;
+import com.dat3m.dartagnan.solver.caat4wmm.coreReasoning.CoreLiteral;
+import com.dat3m.dartagnan.solver.caat4wmm.coreReasoning.CoreReasoner;
+import com.dat3m.dartagnan.utils.logic.Conjunction;
+import com.dat3m.dartagnan.utils.logic.DNF;
+import com.dat3m.dartagnan.verification.VerificationTask;
+import com.dat3m.dartagnan.verification.model.ExecutionModel;
+import org.sosy_lab.java_smt.api.Model;
+import org.sosy_lab.java_smt.api.SolverContext;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/*
+    This is our domain-specific bridging component that specializes the CAATSolver to the WMM setting.
+*/
+public class WMMSolver {
+
+    private final VerificationTask task;
```

Comment:
never used

Reply:
I also was wondering if this field should stay or not. It is not used simply because the constructor already extracts all the components that are used (simply out of convenience rather than necessity). So there are two solutions: Either I only keep the task object and extract the necessary information on demand (so I get rid of the other members) or I always extract all necessary information in the constructor and do not keep the task object.
The former solution causes a lot of boilerplate code but it keeps all information accessible which is helpful for future feature additions. The latter solution causes extra functionality to very likely affect the constructor implementation (every extra feature needs to have the constructor/members changed to extract/store the needed information).

I honestly do not know which solution is more appropriate and that is why I had this middle ground solution where I kept the task object because that is what I effectively want to keep but I also extracted the used components to avoid some boilerplate code.